File: app/src/main/java/app/com/chess/ai/models/global/ChessSquare.java
package app.com.chess.ai.models.global;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class ChessSquare {
    @SerializedName("list")
    public ArrayList<ChessObject> list;

    static public class ChessObject {
        @SerializedName("level")
        public int level;
        @SerializedName("activeSquares")
        public String activeSquares;
        @SerializedName("isUnlocked")
        public boolean isUnlocked;
    }

    public ArrayList<ChessObject> getList() {
        return list;
    }
}

File: app/src/main/java/app/com/chess/ai/models/dbModels/ChessSquare.kt
package app.com.chess.ai.models.dbModels

import com.google.gson.annotations.SerializedName
import java.util.*

class ChessSquare {
    @SerializedName("list")
    var list: ArrayList<ChessObject>? = null

    class ChessObject {
        @SerializedName("level")
        var level = 0

        @SerializedName("activeSquares")
        var activeSquares: String? = null

        @SerializedName("isUnlocked")
        var isUnlocked = false
    }
}